### ISSUE TITLE
Deriving NetworkBehaviour without specifying OutEvent gives compilation error: use of undeclared type `NetworkBehaviour`

### ISSUE THREAD

**Issue by folex** [libp2p/rust-libp2p#2921]
## Summary

The following snippet gives a compilation error
```
use libp2p::identify::Identify;
use libp2p::NetworkBehaviour;
use libp2p::ping::Ping;

#[derive(NetworkBehaviour)]
pub struct Behaviour {
    identify: Identify,
    ping: Ping,
}
```

Here's a repo with that snippet, it reproduces the error on compilation https://github.com/folex/libp2p-network-behaviour-derive-error/tree/master

## Expected behaviour

I expect macro to derive `OutEvent` enum automatically, as specified in doc

> // If we find a `#[behaviour(out_event = "Foo")]` attribute on the
> // struct, we set `Foo` as the out event. If not, the `OutEvent` is
> // generated.

## Actual behaviour

Compilation error
```
error[E0433]: failed to resolve: use of undeclared type `NetworkBehaviour`
 --> src/main.rs:5:10
  |
5 | #[derive(NetworkBehaviour)]
  |          ^^^^^^^^^^^^^^^^ use of undeclared type `NetworkBehaviour`
  |
  = note: this error originates in the derive macro `NetworkBehaviour` (in Nightly builds, run with -Z macro-backtrace for more info)

```

<details><summary>Debug Output</summary>
<p>

```
<output>
```
</p>
</details>

## Possible Solution

Adding an `out_event` with a enum name yields no compilation errors.

## Version
```
libp2p = "0.48.0"
libp2p-core = { version = "0.36.0", default-features = false, features = [ "secp256k1" ] }
libp2p-metrics = { version = "0.9.0", features = ["kad"] }
libp2p-noise = "0.39.0"
libp2p-swarm = "0.39.0"
```

```
$ rustup show
active toolchain
----------------
nightly-x86_64-apple-darwin (default)

$ cargo --version
cargo 1.65.0-nightly (4ed54cecc 2022-08-27)
```

## Would you like to work on fixing this bug?

Maybe

**Comment by elenaf9** [comment]
Thank you for the detailed bug report @folex! 
I believe the issue is that when generating the out event the macro references the [`NetworkBehavour` ](https://docs.rs/libp2p/latest/libp2p/swarm/trait.NetworkBehaviour.html)**trait**, which you don't have imported:
https://github.com/libp2p/rust-libp2p/blob/39d63513d4352ab9acd289bc10fc09d5abd04fcf/swarm-derive/src/lib.rs#L128

So quick fix for you would be to just import the trait from `libp2p::swarm::NetworkBehaviour` (right now you only have the macro itself imported).


That being said, ideally this shouldn't be necessary if we reference the full path in the swarm-derive code like it is done here:
https://github.com/libp2p/rust-libp2p/blob/39d63513d4352ab9acd289bc10fc09d5abd04fcf/swarm-derive/src/lib.rs#L50

Would you be interested in doing a PR with this fix?

**Comment by folex** [comment]
Thanks for the explanation, sorry I didn't see it myself. Name clashing confused me a bunch.

Yep, I will submit a PR today!

**Comment by elenaf9** [comment]
Fixed with #2932. Thanks @folex!
